demos/common/cpp/models/src/hpe_model_associative_embedding.cpp

```diff
@@ -0,0 +1,207 @@
+        throw std::runtime_error("3-channel 4-dimensional model's input is expected");
+    InputInfo& inputInfo = *cnnNetwork.getInputsInfo().begin()->second;
+    inputInfo.setPrecision(Precision::U8);
+    inputInfo.getInputData()->setLayout(Layout::NCHW);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    const OutputsDataMap& outputInfo = cnnNetwork.getOutputsInfo();
+    if (outputInfo.size() != 3)
+        throw std::runtime_error("Demo supports topologies only with 3 outputs");
+    for (const auto& outputLayer: outputInfo) {
+        outputLayer.second->setPrecision(Precision::FP32);
+        outputsNames.push_back(outputLayer.first);
+        const SizeVector& outputLayerDims = outputLayer.second->getTensorDesc().getDims();
+        if (outputLayerDims.size() != 4 && outputLayerDims.size() != 5)
+                throw std::runtime_error("output layers are expected to be 4-dimensional or 5-dimensional");
+        if (outputLayerDims[0] != 1 || outputLayerDims[1] != 17)
+                throw std::runtime_error("output layers are expected to have 1 batch size and 17 channels");
+    }
+}
+
+void HpeAssociativeEmbedding::reshape(CNNNetwork& cnnNetwork) {
+    ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
+    SizeVector& InputLayerDims = inputShapes.begin()->second;
+    if (!targetSize) {
+        targetSize =  static_cast<int>(std::min(InputLayerDims[2], InputLayerDims[3]));
+    }
+    int inputHeight = static_cast<int>(std::round(targetSize * aspectRatio));
+    int inputWidth = targetSize;
+    if (aspectRatio >= 1.0) {
+        std::swap(inputHeight, inputWidth);
+    }
+    int height = static_cast<int>((inputHeight + stride - 1) / stride) * stride;
+    int width = static_cast<int>((inputWidth + stride - 1) / stride) * stride;
+    InputLayerDims[0] = 1;
+    InputLayerDims[2] = height;
+    InputLayerDims[3] = width;
+    inputLayerSize = cv::Size(InputLayerDims[3], InputLayerDims[2]);
+    cnnNetwork.reshape(inputShapes);
+}
+
+std::shared_ptr<InternalModelData> HpeAssociativeEmbedding::preprocess(const InputData& inputData, InferRequest::Ptr& request) {
+    auto& image = inputData.asRef<ImageInputData>().inputImage;
+    cv::Mat resizedImage;
+    float scale = std::min(inputLayerSize.height / static_cast<float>(image.rows),
+                           inputLayerSize.width / static_cast<float>(image.cols));
+    cv::resize(image, resizedImage, cv::Size(), scale, scale, cv::INTER_CUBIC);
+    int h = resizedImage.rows;
+    int w = resizedImage.cols;
+    cv::Mat paddedImage;
+    int bottom = inputLayerSize.height - resizedImage.rows;
+    int right = inputLayerSize.width - resizedImage.cols;
+    cv::copyMakeBorder(resizedImage, paddedImage, 0, bottom, 0, right,
+                       cv::BORDER_CONSTANT, meanPixel);
+    Blob::Ptr frameBlob = request->GetBlob(inputsNames[0]);
+    matU8ToBlob<uint8_t>(paddedImage, frameBlob);
```

Comment:
1. Can you replace `matU8ToBlob()` with `wrapMat2Blob()` and set input layout to `NHWC` for both hpe wrappers

2. It looks that `wrapMat2Blob()` doesn't ship ownership of data to `Blob`, so we actually should store `cv::Mat` in `InternalModelData`, but given that other demos have that problem, let's leave it as is for now.